crates/turborepo-lib/src/daemon/proto/turbod.proto

```diff
@@ -92,6 +94,18 @@ message DiscoverPackagesRequest {
 
 }
 
+message PackageChangesRequest {}
+
+message PackageChangeEvent {
+  PackageChangeType change_type = 1;
```

Comment:
Would it be helpful to use [oneof](https://protobuf.dev/programming-guides/proto3/#oneof) here? You can either send a rediscover everything message, or a package change, even with an optional name.

Reply:
Hmm I guess we could do something like:

```
oneof change_oneof {
  bool rediscover = 1;
  string package_name = 2;
}
```

But that wouldn't be extendable if we wanted to add more message types, and the boolean is a little awkward there. Is there a better way? I'm not super familiar with protobuf

Reply:
What about something like this:
```
message PackageChanged {
  string package_name = 1;
}

message RediscoverPackages {}

message PackageChangeEvent {
  oneof event {
    PackageChanged package_changed = 1;
    RediscoverPackages rediscover_packages = 2;
  };
}
```

Reply:
Ah yeah that makes a lot more sense. Thanks!